recipes/_nfs_config.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: _nfs_config
+#
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NFS cookbook server_template attribute is not updated for centos8, provide the correct override
+node.force_override['nfs']['config']['server_template'] = value_for_platform(
+  ['centos'] => {
+    '>=8' => '/etc/nfs.conf'
+  },
+  'ubuntu' => { 'default' => '/etc/default/nfs-kernel-server' },
+  'default' => '/etc/sysconfig/nfs'
+)
+
+# For performance, set NFS threads to max(num_cores, 8); note cores, not vcpus
+ruby_block "set nfs threads" do
+  block do
+    require 'chef/mixin/shell_out'
+    cores_per_socket = shell_out!('lscpu | grep -oP "^Core\(s\) per socket:\s*\K(.+)$"').stdout.strip.to_i
+    sockets = shell_out!('lscpu | grep -oP "^Socket\(s\):\s*\K(.+)$"').stdout.strip.to_i
+    total_cores = cores_per_socket * sockets
+    node.force_override['nfs']['threads'] = [total_cores, 8].max
+  end
+end
+
+if node['platform'] == 'centos' && node['platform_version'].to_i == 8
+  # Workaround for issue: https://github.com/atomic-penguin/cookbook-nfs/issues/116
+  node.force_override['nfs']['service']['idmap'] = 'nfs-idmapd'
+end
+
+if node['platform'] == 'ubuntu' && node['platform_version'].to_f >= 16.04
+  # FIXME: https://github.com/atomic-penguin/cookbook-nfs/issues/93
+  include_recipe "nfs::server"
+end
+include_recipe "nfs::server4"
+
+# Extend NFS cookbook logic to OS not officially supported
+# By overwriting templates for node['nfs']['config']['server_template'] used by NFS cookbook
+# When running, NFS cookbook will use nfs.conf.erb templates provided in this cookbook to generate server_template
+edit_resource(:template, node['nfs']['config']['server_template']) do
```

Comment:
don't we want to override only the missing/broken templates?

Reply:
Sure, added logic to only overwrite for alinux OSs and centos8